### PR TITLE
Height of shading objects is incorrect

### LINKED ISSUE #4764: floor_to_ceiling_height for Shading objects is not respected in the floorspace reverse translator
Issue overview
--------------
The shading groups created by the floorspace reverse translator do not have the correct height

## Current Behavior
A shading group is created for each shading object in the floor space json file, a collection of faces is made by extruding the 'footprint' of the shading object. The extrusion height is specified by floor_to_ceiling_height. Currently all shading groups are created with an extrusion length of 10' regardless of the value of floor_to_ceiling_height 

## Expected Behavior
The extrusion shape should be created with the height as specified in the floorspace json file

## Steps to Reproduce

1. Generate an OSM model using the following file (can't upload a json file but it is the file referred to in #4323 ) via the floorspace reverse translator
2. Do the same using the ThreeJS reverse translator
3. Look at shading group 1-10 (for example)

Model as generated by the ThreeJS reverse translator
image

Model as generated by the floorspace reverse translator
image

## Possible Solution
Will have to correct the floorspace reverse translator code that supplies the extrusion height

## Details
I have a floorspace reverse translator test that verifies this issue

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version):
 - Version of OpenStudio (if using an intermediate build, include SHA):
latest version of develop at commit 351d37fb

### PR DESCRIPTION
**Pull request by ggartside**
Fixes 4664: floor_to_ceiling_height for Shading objects is not respected in the floorspace reverse translator

This what the model should look like
image

and this is what the model currently looks like
image

Pull request overview
---------------------

 - Fixes #4764

Fixes issue 4764 - Shading Floorspace objects are not converted to ShadingGroups correctly when the Shading object specifies an override for the height of the shading object 

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [x] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by jmarrec** [comment]
@ggartside To facilitate review, could you try to separate your changes into distinct commits, ideally in a TDD fashion (add test first, then fix, you can always rebase and reorder commits at the end)

* Add a test
* (do renames)
* fix

I have done a rebase and reauthored commits to attribute them to you this time for convenience.

Also, could you post a picture showing the before/after like you did in https://github.com/NREL/OpenStudio/issues/4764 ?

File: src/model/test/FloorspaceReverseTranslator_GTest.cpp [hunk 1]
```diff
@@ -82,6 +82,35 @@ using namespace openstudio::model;
     OutputDebugString(os_.str().c_str()); \
   }
 
+void CompareSurfaceGroups(Model& model, Model& baseline) {
+
+  auto shadingGroup1 = model.getModelObjects<ShadingSurfaceGroup>();
+  auto shadingGroup2 = baseline.getModelObjects<ShadingSurfaceGroup>();
+  EXPECT_EQ(shadingGroup1.size(), shadingGroup2.size());
```
Comment (jmarrec) [inline src/model/test/FloorspaceReverseTranslator_GTest.cpp]: Before fix, the `FloorspaceReverseTranslator_shadingtest` reports a failure here, is that expected?

```
2524: Expected equality of these values:
2524:   shadingGroup1.size()
2524:     Which is: 12
2524:   shadingGroup2.size()
2524:     Which is: 11
```

`FloorspaceReverseTranslator_Issue_4764` reports equality. Both reports lots of failure on line 108 where matched is false.

File: src/model/test/FloorspaceReverseTranslator_GTest.cpp [hunk 2]
```diff
@@ -82,6 +82,35 @@ using namespace openstudio::model;
...
+  for (auto& shadingGroup : shadingGroup1) {
+    // Shading Surface Group names only match for building shading
+    if (shadingGroup.shadingSurfaceType() == "Building") {
+      auto match = baseline.getModelObjectByName<ShadingSurfaceGroup>(*shadingGroup.name());
+      EXPECT_TRUE(match.has_value());
```
Comment (jmarrec) [inline src/model/test/FloorspaceReverseTranslator_GTest.cpp]: FYI it's possible to Improve reporting, for eg here:

```suggestion
      auto shadingGroupName = shadingGroup.nameString();
      auto match = baseline.getModelObjectByName<ShadingSurfaceGroup>(shadingGroupName);
      EXPECT_TRUE(match.has_value()) << "Could not locate shading group " << shadingGroupName;
```

File: src/model/test/FloorspaceReverseTranslator_GTest.cpp (hunk 1 above)
Comment (jmarrec) [reply src/model/test/FloorspaceReverseTranslator_GTest.cpp]: If it is, then I'm all good with this PR!

Comment (jmarrec) [reply src/model/test/FloorspaceReverseTranslator_GTest.cpp]: @ggartside this is the last item that needs addressing before merging please. Is this expected or not?

File: src/model/test/FloorspaceReverseTranslator_GTest.cpp (hunk 2 above)
Comment (ggartside) [reply src/model/test/FloorspaceReverseTranslator_GTest.cpp]: Thanks - yeah I sort of discovered that bay accident a while back so have been using it on other branches

File: src/model/test/FloorspaceReverseTranslator_GTest.cpp (hunk 1 above)
Comment (ggartside) [reply src/model/test/FloorspaceReverseTranslator_GTest.cpp]: I ran the test here and it passes and it looks like all the tests pass. The shading groups should match.

**Pull request merged**